common/src/main/java/com/wynntils/models/activities/mapdata/ActivityProvider.java

```diff
@@ -0,0 +1,99 @@
+import com.wynntils.services.mapdata.attributes.impl.AbstractMapLocationAttributes;
+import com.wynntils.services.mapdata.attributes.type.MapLocationAttributes;
+import com.wynntils.services.mapdata.features.impl.MapLocationImpl;
+import com.wynntils.services.mapdata.features.type.MapFeature;
+import com.wynntils.services.mapdata.providers.builtin.BuiltInProvider;
+import com.wynntils.utils.colors.CustomColor;
+import com.wynntils.utils.mc.type.Location;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+public class ActivityProvider extends BuiltInProvider {
+    private static final String ACTIVITY_LOCATION_NAME = "Activity Location";
+
+    private ActivityLocation spawnLocation;
+    private ActivityLocation trackedActivityLocation;
+
+    public void setSpawnLocation(Location spawnLocation) {
+        if (spawnLocation == null) {
+            this.spawnLocation = null;
+        } else {
+            this.spawnLocation = new ActivityLocation(ACTIVITY_LOCATION_NAME, ActivityType.QUEST, spawnLocation);
+        }
+    }
+
+    public void setTrackedActivityLocation(Location trackedActivityLocation, CustomColor activityColor) {
+        if (trackedActivityLocation == null || activityColor == null) {
+            this.trackedActivityLocation = null;
+        } else {
+            this.trackedActivityLocation = new ActivityLocation(
+                    Models.Activity.getTrackedName(), Models.Activity.getTrackedType(), trackedActivityLocation);
+        }
+    }
+
+    @Override
+    public String getProviderId() {
+        return "activities";
+    }
+
+    @Override
+    public void reloadData() {
+        // This built-in provider does not load any external data, so no explicit reload is needed.
+    }
+
+    @Override
+    public Stream<MapFeature> getFeatures() {
+        Stream.Builder<MapFeature> features = Stream.builder();
+
+        if (trackedActivityLocation != null) {
+            features.add(trackedActivityLocation);
+        }
+
+        if (spawnLocation != null
+                && (trackedActivityLocation != null
+                        && !spawnLocation.getLocation().equals(trackedActivityLocation.getLocation()))) {
+            features.add(spawnLocation);
+        }
+
+        return features.build().filter(Objects::nonNull);
+    }
+
+    public static class ActivityLocation extends MapLocationImpl {
+        private final String name;
+        private final ActivityType activityType;
+
+        public ActivityLocation(String name, ActivityType activityType, Location location) {
+            super(
+                    "activity-"
+                            + (name.isBlank()
+                                    ? location.hashCode()
+                                    : name.replaceAll(" +", "-").toLowerCase(Locale.ROOT)),
+                    "wynntils:activity:" + CaseFormat.UPPER_UNDERSCORE.to(CaseFormat.LOWER_HYPHEN, activityType.name()),
```

Comment:
These should actually be some util of some sort, eg. MapDataUtils#sanitzeFeature/Category